client/src/partials/patients/registry/modal.js

```diff
@@ -0,0 +1,100 @@
+angular.module('bhima.controllers')
+.controller('PatientRegistryModalController', PatientRegistryModalController);
+
+PatientRegistryModalController.$inject = [
+  '$uibModalInstance', 'InventoryService', 'PatientService', 'util'
+];
+
+function PatientRegistryModalController( $uibModalInstance, Inventory, patients, util) {
+  var vm = this;
+
+  vm.period = {
+    today : {
+      cacheKey : 'today',
+      translateKey : 'FORM.BUTTONS.TODAY',
+      id : 'today'
+    },
+    week : {
+      cacheKey : 'week',
+      translateKey : 'FORM.BUTTONS.THIS_WEEK',
+      id : 'week'
+    },
+    month : {
+      cacheKey : 'month',
+      translateKey : 'FORM.BUTTONS.THIS_MONTH',
+      id : 'month'
+    },
+    year : {
+      cacheKey : 'year',
+      translateKey : 'FORM.BUTTONS.THIS_YEAR',
+      id : 'year'
+    }
+  };  
+
+  // bind methods
+  vm.submit = submit;
+  vm.cancel = cancel;
+  vm.setTimes = setTimes;
+  vm.today = new Date();
+
+
+  vm.data = { is_percentage : 0 };
+
+  function submit(form) {
+    if (form.$invalid) { return; }
+    vm.patient = util.clean(vm.patient);
+
+    var patient = angular.copy(vm.patient);
+    patient.detail = 1;
+    
+    var promise = patients.search(patient);
+    var patientFilters = patients.patientFilters(patient);
+
+    promise
+    .then(function (response) {   
+      var data = {
+        response : response,
+        filters   : patientFilters 
+      };
+      
+      return $uibModalInstance.close(data);
+
+    });    
+  }
+
+  function setTimes(times){
```

Comment:
I think these methods already exist in the `DateService`. They would look like this:
```js

PatientRegistryModalController.$inject = [
  /* other injects ..., */ 'DateService'
];

function PatientRegistryModalController(/* injects ,*/ Dates) {

  // controller code ..

  // set the duration of the query
  function setTimes(times) {

   // the dateRegistrationTo never changes, so we set it at the beginning
   vm.patient = {
     dateRegistrationTo : new Date()
   };

   switch (times) {
      case 'today' :
        vm.patient.dateRegistrationFrom = new Date();
        break;
      case 'week' :
        vm.patient.dateRegistrationFrom = Dates.previous.week();
        break;
      case 'month' :
        vm.patient.dateRegistrationFrom = Dates.previous.month();
        break;
      default:
        vm.patient.dateRegistrationFrom = Dates.previous.year();
  }
}
```